massa-ledger-worker/src/ledger_db.rs

```diff
@@ -184,19 +197,52 @@ impl LedgerDB {
         self.db.write(batch).expect(CRUD_ERROR);
     }
 
-    /// Set the disk ledger metadata
+    /// Set the disk ledger slot metadata
     ///
     /// # Arguments
     /// * slot: associated slot of the current ledger
     /// * batch: the given operation batch to update
-    ///
-    /// NOTE: right now the metadata is only a Slot, use a struct in the future
-    fn set_metadata(&self, slot: Slot, batch: &mut WriteBatch) {
+    fn set_slot(&self, slot: Slot, batch: &mut WriteBatch, ledger_hash: &mut Hash) {
         let handle = self.db.cf_handle(METADATA_CF).expect(CF_ERROR);
-        let mut bytes = Vec::new();
+        let mut slot_bytes = Vec::new();
         // Slot serialization never fails
-        self.slot_serializer.serialize(&slot, &mut bytes).unwrap();
-        batch.put_cf(handle, SLOT_KEY, bytes);
+        self.slot_serializer
+            .serialize(&slot, &mut slot_bytes)
+            .unwrap();
+        batch.put_cf(handle, SLOT_KEY, slot_bytes.clone());
+        // XOR previous slot and new one
+        if let Some(prev_bytes) = self.db.get_cf(handle, SLOT_KEY).expect(CRUD_ERROR) {
```

Comment:
Maybe it's worth adding an identifier to the slot metadata bytes so that the hash can't collide with another one of metadata in the future

Reply:
Bytes of slot are just two varint we can imagine an other metadata having the same format (or close) that could lead to having the same hash. Similar as https://github.com/massalabs/massa/discussions/2968#discussioncomment-3571775